### PR TITLE
fix(nx): check for valid semver version before updating package

### LINKED ISSUE #1831: migrations for package '@nrwl/angular' cause an error
_[Please make sure you have read the submission guidelines before posting an issue](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#-submitting-issue)_

## Expected Behavior

when running `ng update @nrwl/angular @nrwl/workspace` the migration should work or emit better error in the log.

e.g: Invalid Version: ^8.3.3, Version x.x.x required of package x

## Current Behavior

running `ng update @nrwl/angular @nrwl/workspace` cause an error:
```
** Executing migrations for package '@nrwl/angular' **
Invalid Version: ^8.3.3
```
I'm updating from:
```
Name                                  Version                  Command to update
-----------------------------------------------------------------------------------
@angular/cdk                          8.0.2 -> 8.2.0           ng update @angular/cdk
@angular/cli                          8.3.3 -> 8.3.4           ng update @angular/cli
@nrwl/angular                         8.2.0 -> 8.5.0           ng update @nrwl/angular
@nrwl/workspace                       8.2.0 -> 8.5.0           ng update @nrwl/workspace
```
Same error even if I update angilar/cli to latest before updating @nrwl/...

### PR DESCRIPTION
**Pull request by brandonroberts**
_[Please make sure you have read the submission guidelines before posting an PR](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#submit-pr)_

> _Please make sure that your commit message follows our format._

> Example: `fix(nx): must begin with lowercase`

## Current Behavior (This is the behavior we have today, before the PR is merged)

Semver check fails if version contains a range, such as `^8.0.0`

## Expected Behavior (This is the new behavior we can expect after the PR is merged)

Valid versions will pass
`8.0.0`
`^8.0.0`
`~8.0.0`

If the version cannot be considered `clean` by `semver`, it will fail
`>=8.0.0`

## Issue

Closes #1831

### CODE REVIEW

File: packages/workspace/src/utils/version-utils.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  if (clean(newVersion)) {
+    return newVersion;
+  }
+
+  if (version.startsWith('~') || version.startsWith('^')) {
+    newVersion = version.substring(1);
+  }
+
+  if (!clean(newVersion)) {
```
Comment (jdpearce) [inline packages/workspace/src/utils/version-utils.ts]: You're using `clean` but not actually returning the cleaned result. Did you mean to use `valid` instead?

File: packages/workspace/src/utils/version-utils.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { clean } from 'semver';
+
+export function checkAndCleanWithSemver(version: string) {
+  let newVersion = version;
+
+  if (clean(newVersion)) {
```
Comment (jdpearce) [inline packages/workspace/src/utils/version-utils.ts]: Did you mean to use `valid` here?

File: packages/workspace/src/utils/version-utils.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    return newVersion;
+  }
+
+  if (version.startsWith('~') || version.startsWith('^')) {
+    newVersion = version.substring(1);
+  }
+
+  if (!clean(newVersion)) {
+    throw new Error(
+      `Unable to validate the provided version against semver - (${version})`
```
Comment (jdpearce) [inline packages/workspace/src/utils/version-utils.ts]: This is a pretty unfriendly error message. Is it worth catching this in the migration and logging out something friendlier like "Your package.json lists a version of @angular/cli that Nx is unable to validate - (${version})"?

File: packages/angular/src/migrations/update-8-5-0/update-8-5-0.spec.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
+import { Tree } from '@angular-devkit/schematics';
+import { readJsonInTree } from '@nrwl/workspace';
+import {
+  SchematicTestRunner,
+  UnitTestTree
+} from '@angular-devkit/schematics/testing';
+import { serializeJson } from '@nrwl/workspace';
+import { join } from 'path';
+
+describe('Update 8.3.0', () => {
```
Comment (jdpearce) [inline packages/angular/src/migrations/update-8-5-0/update-8-5-0.spec.ts]: TESTS! AWESOME! 🙏

**Review by catfireparty — CHANGES_REQUESTED** [review]

**Comment by jdpearce** [comment]
Sorry, this looks to be failing on the jest builder spec that I modified. I have no idea how the failing tests managed to get into master... 😞

**Pull request merged**